core-default/src/main/java/me/aurium/beetle/defaults/file/LocalSourceProvisioner.java

```diff
@@ -0,0 +1,41 @@
+package me.aurium.beetle.defaults.file;
+
+import me.aurium.beetle.api.datacore.CoreSource;
+import me.aurium.beetle.api.file.UncheckedIOException;
+import me.aurium.beetle.api.file.producer.ProvisionAccessor;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+public class LocalSourceProvisioner implements ProvisionAccessor {
+
+    private final StateHolder<CoreSource> stateHolder;
+
+    private final Path fullPath;
+    private final LocalSourceOptions options;
+
+
+    public LocalSourceProvisioner(StateHolder<CoreSource> stateHolder, Path fullPath, LocalSourceOptions options) {
+        this.stateHolder = stateHolder;
+        this.fullPath = fullPath;
+        this.options = options;
+    }
+
+    @Override
+    public void load() {
+        File file = fullPath.toFile();
+
+        try {
+            if (!file.createNewFile()) {
+                throw new IOException("Database file already exists!");
```

Comment:
```suggestion
                throw new UncheckedIOException("Database file already exists!");
```
It's somewhat of a rule to not throw the exception you are catching inside the try block because doing so suggests exceptions being used as control flow.